packages/dashboard-frontend/src/containers/FactoryLoader/index.tsx

```diff
@@ -458,6 +463,67 @@ export class FactoryLoaderContainer extends React.PureComponent<Props, State> {
     return workspace;
   }
 
+  private async createDevWorkspaceFromPrebuiltResources(
+    devWorkspacePrebuiltResources: string,
+    factoryParams: string,
+  ): Promise<Workspace | undefined> {
+    let workspace: Workspace | undefined;
+
+    // creation policy is `peruser`
+    workspace = this.props.allWorkspaces.find(workspace => {
+      if (isCheWorkspace(workspace.ref)) {
+        return false;
+      } else {
+        const annotations = workspace.ref.metadata.annotations;
+        const source = annotations ? annotations[DEVWORKSPACE_DEVFILE_SOURCE] : undefined;
+        if (source) {
+          const sourceObj = safeLoad(source) as FactorySource;
+          return sourceObj?.factory?.params === factoryParams;
+        }
+        return false;
+      }
+    });
+
+    if (!workspace) {
+      try {
+        let yamlContent: string;
+        try {
+          const response = await axios.get(devWorkspacePrebuiltResources);
+          yamlContent = response.data;
+        } catch (e) {
+          const errorMessage = common.helpers.errors.getMessage(e);
+          console.error(
+            `Failed to fetch prebuilt resources from ${devWorkspacePrebuiltResources}. ${errorMessage}`,
+          );
+          this.showAlert(`Failed to fetch prebuilt resources. ${errorMessage}`);
+          return;
+        }
+
+        const resources = safeLoadAll(yamlContent);
+        const devworkspace = resources.find(
+          resource => resource.kind === 'DevWorkspace',
+        ) as devfileApi.DevWorkspace;
+        const devworkspaceTemplate = resources.find(
+          resource => resource.kind === 'DevWorkspaceTemplate',
+        ) as devfileApi.DevWorkspaceTemplate;
+
+        await this.props.createWorkspaceFromPrebuiltResources(devworkspace, devworkspaceTemplate);
+
```

Comment:
With this solution, we will be able to create only one workspace from a sample.

But, as a developer, I prefer to have 'a quick start' flow, when I can choose a sample with needed technology,  create a new workspace. Then, add a new project inside, push it to upstream without requirement to recreate a new workspace. 

Note: 
We  need to recreate workspace, because with this implementation click on a sample creates a new devworkspace  from prebuilt resources without changing its name(adding some generated symbols as it was previosly). 
And we can't create two workspaces with different name from the same sample, can't rename existing devworkspace. So, it will require to recreate each workspace wich was based on that sample.

My proposal: improve this implementation and add some generated symbols to devworkspace and devworkspacetemplate names

WDYT?

Reply:
Good point, thanks!
Actually, I don't have any strong opinion on how do we proceed with this. 
@ibuziuk @l0rd can you please share your thoughts?

Reply:
> With this solution, we will be able to create only one workspace from a sample.

I believe this is fine for the first iteration. Basically,  if a workspace based on the sample already exists, clicking on the devfile sample again will reopen the existing one. For the time being I would avoid doing any modifications of the DevWorkspace / DevWorkspace template on UD level